### PR TITLE
[IMP] Adding new check subpackage-conflict-with-python-file

### PR DESCRIPTION
**Pull request by JesusZapata**
Conflict on the resolution of names given that the folder in the package and a file in the package are called equal

### Fixes / new features
- New Feature

Fix https://github.com/PyCQA/pylint/issues/1038

### CODE REVIEW

**Comment by JesusZapata** [comment]
Create folder to test

```bash
mkdir -p /tmp/my_wrong_package/my_wrong_subpackage/
touch /tmp/my_wrong_package/{.,my_wrong_subpackage}/__init__.py
touch /tmp/my_wrong_package/my_wrong_subpackage.py
```
And run pylint

```
python -m pylint -rn -d all -e subpackage-conflict-with-python-file /tmp/my_wrong_package/
```

The output is 

```
************* Module my_wrong_package.my_wrong_subpackage
R:  1, 0: Sub package folder (tmp/my_wrong_package/my_wrong_subpackage) conflict with sub package python file (tmp/my_wrong_package/my_wrong_subpackage.py) (subpackage-conflict-with-python-file)
```

**How do I test for this feature?**

**Comment by rogalski** [comment]
What about a message name `package-conflicts-with-module`? Or maybe something along lines of `package-module-conflict` (just to avoid suggesting that one is more preferable to other)?

Packages and modules (and difference between them) is well documented, sub- prefix may be misleading (it's not _always_ a subpackage), and `python` in message name does not give any meaningful information to user.

About testing it, I'm not quite sure yet. There is an old infrastructure of `test_func`, but it has [explicit assertion for not adding additional tests to it](https://github.com/PyCQA/pylint/blob/master/pylint/test/test_func.py#L52). @PCManticore what do you think about adding a capability of handling packages as well as modules in `test_functional`? May also help with porting these old tests to new framework.

**Comment by PCManticore** [comment]
I think ``test_functional`` is overcomplicated as is. Testing with packages should be relatively easy if done through a normal test, such as this one https://github.com/PyCQA/pylint/blob/3d14bcb35e7fd9b744be353cf6ca689b1ea038b3/pylint/test/unittest_checker_imports.py#L82-L82. The old tests weren't ported yet not due to lack of features on ``test_functional``'s side, but due to not having time and being a very tedious, not important task.

**Comment by JesusZapata** [comment]
@PCManticore @rogalski @moylop260 
Thanks for the help
Now all checks is in green

File: pylint/checkers/refactoring.py [hunk 1]
```diff
@@ -102,6 +103,11 @@ class RefactoringChecker(checkers.BaseTokenChecker):
                   'weird bugs in your code. You should always use parentheses '
                   'explicitly for creating a tuple.',
                   {'minversion': (3, 0)}),
+        'R1708': ('Sub package folder (%s) conflict with sub package python '
```
Comment (rogalski) [inline pylint/checkers/refactoring.py]: Message seems to be left-over from previous revision. We decided to drop sub- prefixes.

Comment (JesusZapata) [inline pylint/checkers/refactoring.py]: @rogalski 
I made the modify on the message
Are you agree?

**Comment by PCManticore** [comment]
Closing this. While the check could be useful, I doubt it is so important to have it in core pylint. Another suggestion would be to have this as a third party library that folks could install on their own.

**Pull request closed without merging**